src/app/clusters/group-key-mgmt-server/group-key-mgmt-server.cpp

```diff
@@ -267,49 +267,54 @@ bool emberAfGroupKeyManagementClusterKeySetWriteCallback(
         return true;
     }
 
-    if (commandData.groupKeySet.epochKey0.empty() || (0 == commandData.groupKeySet.epochStartTime0))
+    if (commandData.groupKeySet.epochKey0.IsNull() || commandData.groupKeySet.epochStartTime0.IsNull() ||
+        commandData.groupKeySet.epochKey0.Value().empty() || (0 == commandData.groupKeySet.epochStartTime0.Value()))
     {
         // If the EpochKey0 field is null or its associated EpochStartTime0 field is null,
         // then this command SHALL fail with an INVALID_COMMAND
         emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_INVALID_COMMAND);
         return true;
     }
 
-    GroupDataProvider::KeySet keyset(commandData.groupKeySet.groupKeySetID, commandData.groupKeySet.securityPolicy, 0);
+    GroupDataProvider::KeySet keyset(commandData.groupKeySet.groupKeySetID, commandData.groupKeySet.groupKeySecurityPolicy, 0);
 
     // Epoch Key 0
-    keyset.epoch_keys[0].start_time = commandData.groupKeySet.epochStartTime0;
-    memcpy(keyset.epoch_keys[0].key, commandData.groupKeySet.epochKey0.data(), GroupDataProvider::EpochKey::kLengthBytes);
+    keyset.epoch_keys[0].start_time = commandData.groupKeySet.epochStartTime0.Value();
+    memcpy(keyset.epoch_keys[0].key, commandData.groupKeySet.epochKey0.Value().data(), GroupDataProvider::EpochKey::kLengthBytes);
     keyset.num_keys_used++;
 
     // Epoch Key 1
-    if (!commandData.groupKeySet.epochKey1.empty())
+    if (!commandData.groupKeySet.epochKey1.IsNull() && !commandData.groupKeySet.epochStartTime1.IsNull() &&
+        !commandData.groupKeySet.epochKey1.Value().empty())
     {
-        if (commandData.groupKeySet.epochStartTime1 <= commandData.groupKeySet.epochStartTime0)
+        if (commandData.groupKeySet.epochStartTime1.Value() <= commandData.groupKeySet.epochStartTime0.Value())
         {
             // If the EpochKey1 field is not null, its associated EpochStartTime1 field SHALL contain
             // a later epoch start time than the epoch start time found in the EpochStartTime0 field.
             emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_INVALID_COMMAND);
             return true;
         }
-        keyset.epoch_keys[1].start_time = commandData.groupKeySet.epochStartTime1;
-        memcpy(keyset.epoch_keys[1].key, commandData.groupKeySet.epochKey1.data(), GroupDataProvider::EpochKey::kLengthBytes);
+        keyset.epoch_keys[1].start_time = commandData.groupKeySet.epochStartTime1.Value();
+        memcpy(keyset.epoch_keys[1].key, commandData.groupKeySet.epochKey1.Value().data(),
+               GroupDataProvider::EpochKey::kLengthBytes);
         keyset.num_keys_used++;
     }
 
     // Epoch Key 2
-    if (!commandData.groupKeySet.epochKey2.empty())
+    if (!commandData.groupKeySet.epochKey2.IsNull() && !commandData.groupKeySet.epochStartTime2.IsNull() &&
```

Comment:
Followup:
* If epochKey2 is not null but epochKey1 is null, should fail per spec.
* If epochKey2 is not null but epochStartTime2 is null, should fail per spec.

Reply:
Fixed.